web_ui/frontend/components/Namespace.tsx

```diff
@@ -0,0 +1,295 @@
+            </Box>
+        </Box>
+    )
+}
+
+export const NamespaceCardSkeleton = () => {
+    return <Skeleton variant="rounded" width={"100%"} height={60} />
+}
+
+interface PendingCardProps {
+    namespace: Namespace;
+    onUpdate: () => void;
+    onAlert: (alert: Alert) => void;
+    authenticated?: Authenticated
+}
+
+export const PendingCard = ({
+    namespace,
+    onUpdate,
+    onAlert,
+    authenticated
+}: PendingCardProps) => {
+
+    const ref = useRef<HTMLDivElement>(null);
+    const [transition, setTransition] = useState<boolean>(false);
+
+    const approveNamespace = async (e: React.MouseEvent) => {
+        try {
+            const response = await secureFetch(`/api/v1.0/registry_ui/namespaces/${namespace.id}/approve`, {
+                method: "PATCH"
+            })
+
+            if (!response.ok){
+                onAlert({severity: "error", message: `Failed to approve namespace: ${namespace.prefix}`})
+            } else {
+                onUpdate()
+                onAlert({severity: "success", message: `Successfully approved namespace: ${namespace.prefix}`})
+            }
+
+        } catch (error) {
+            console.error(error)
+        } finally {
+            e.stopPropagation()
+        }
+    }
+
+    const denyNamespace = async (e: React.MouseEvent) => {
+        try {
+            const response = await secureFetch(`/api/v1.0/registry_ui/namespaces/${namespace.id}/deny`, {
+                method: "PATCH"
+            })
+
+            if (!response.ok){
+                onAlert({severity: "error", message: `Failed to deny namespace: ${namespace.prefix}`})
+            } else {
+                onUpdate()
+                onAlert({severity: "success", message: `Successfully denied namespace: ${namespace.prefix}`})
+            }
+
+        } catch (error) {
+            console.error(error)
+        } finally {
+            e.stopPropagation()
+        }
+    }
+
+    return (
+        <Box>
+            <Box sx={{
+                    cursor: "pointer",
+                    display: "flex",
+                    width: "100%",
+                    justifyContent: "space-between",
+                    border: "solid #ececec 1px",
+                    borderRadius: transition ? "10px 10px 0px 0px" : 2,
+                    "&:hover": {
+                        bgcolor: "#ececec"
+                    },
+                    p: 1
+                }}
+                bgcolor={"secondary"}
+                onClick={() => setTransition(!transition)}
+            >
+                <Box my={"auto"} ml={1}>
+                    <Typography>{namespace.prefix}</Typography>
+                </Box>
+                <Box>
+                    { authenticated?.role == "admin" &&
+                        <>
+                            <Tooltip title={"Deny Registration"}>
+                                <IconButton sx={{bgcolor: "#ff00001a", mx: 1}} color={"error"} onClick={denyNamespace}><Block/></IconButton>
```

Comment:
When click the approve button, admin metadata will also be expanded. May be call `stopPropagation()` on those buttons?

Reply:
I do, I will look into why it isn't working.